File: PopularMovies/src/main/java/com/mchapagai/movies/api/MovieAPI.java
package com.mchapagai.movies.api;

import com.mchapagai.core.response.common.ReviewListResponse;
import com.mchapagai.core.response.common.VideoListResponse;
import com.mchapagai.core.response.movies.MovieCreditResponse;
import com.mchapagai.core.response.movies.MovieDetailsResponse;
import com.mchapagai.core.response.movies.MovieListResponse;
import com.mchapagai.movies.model.binding.CombinedPersonResponse;
import com.mchapagai.movies.model.binding.PersonResponse;

import io.reactivex.Flowable;
import io.reactivex.Observable;
import io.reactivex.Single;

public interface MovieAPI {

    Flowable<MovieListResponse> discoverMovies(int page, String sortBy);

    Observable<VideoListResponse> getMovieVideosbyId(int movieId);

    Observable<ReviewListResponse> getMovieReviewsById(int movieId);

    Observable<MovieDetailsResponse> getMovieDetails(int movieId);

    Observable<MovieCreditResponse> getMovieCreditDetails(int movieId);

    Single<PersonResponse> getPersonDetailsById(int personId);

    Observable<CombinedPersonResponse> getPersonCombinedDetailsById(int personId);

    Observable<MovieListResponse> searchMovies(String query);
}

File: Core/src/main/java/com/mchapagai/core/api/MovieAPI.kt
package com.mchapagai.core.api

import com.mchapagai.core.response.common.ReviewListResponse
import com.mchapagai.core.response.common.VideoListResponse
import com.mchapagai.core.response.movies.MovieCreditResponse
import com.mchapagai.core.response.movies.MovieDetailsResponse
import com.mchapagai.core.response.movies.MovieListResponse
import io.reactivex.Flowable
import io.reactivex.Observable

interface MovieAPI {
    fun discoverMovies(page: Int, sortBy: String): Flowable<MovieListResponse>
    fun fetchMovieDetailsByMovieId(movieId: Int): Observable<MovieDetailsResponse>
    fun getMovieCreditDetailsByCreditId(movieId: Int): Observable<MovieCreditResponse>
    fun getMovieVideosById(movieId: Int): Observable<VideoListResponse>
    fun getMovieReviewsById(movieId: Int): Observable<ReviewListResponse>
}
